Modules/_operator.c

```diff
@@ -1549,16 +1549,90 @@ static PyType_Spec attrgetter_type_spec = {
 typedef struct {
     PyObject_HEAD
     PyObject *name;
-    PyObject *args;
+    PyObject *xargs; // reference to arguments passed in constructor
     PyObject *kwds;
+    PyObject **vectorcall_args;  /* Borrowed references */
+    PyObject *vectorcall_kwnames;
+    vectorcallfunc vectorcall;
 } methodcallerobject;
 
+static void * _methodcaller_initialize_vectorcall(methodcallerobject* mc)
```

Comment:
```suggestion
static int _methodcaller_initialize_vectorcall(methodcallerobject* mc)
```

Reply:
Returning int gives many warnings, because `NULL` is typed as `void *`. For example `return PyErr_NoMemory();` returns `PyObject *` (with value 0). Should the return type really be int?

Reply:
You don't have to return `PyErr_NoMemory()` just return status code after calling the `PyErr_NoMemory()`

https://github.com/python/cpython/blob/a24e25d74bb5d30775a61853d34e0bb5a7e12c64/Modules/atexitmodule.c#L32C1-L35